### DISCUSSION TITLE
Physical Cargo

### DISCUSSION THREAD

**Discussion by ZachZent** [tsunamayo/Starship-EVO#4088] [Suggestions]
To have an open ended discussion on record, how would everyone suggest physical cargo be handled? From what I have read, it doesn't look like Tsuna has come to a major decision, granted it is still early on and isn't a priority right now. Maybe by having a non-live chat discussion on how each thinks it should be, a more though out idea can be formed.

**Comment by ZachZent** [comment]
To start with the oldest (recorded) one, https://github.com/tsunamayo/Starship-EVO/issues/1292

In a nutshell, there would be premade "chests" of preset sizes. Think Minecraft chests. Along with these chests would be a "shelf" block. You would place them down and chests can be placed into the number of slots. Think if you have a stretched 10m long and 10m high shelf, you can stack 100 1x1m chests inside. Although scaling wasn't a thing back then, perhaps scaling will determine what size of chest can be placed. 4x scale can hold 4x4m chests.

The advantage of these shelves is it offers a predefined and physical location for both players and NPCs to access and pickup their stored chests in an easy and predictable way. The shelves being their own unit will allow for their individual manifest both in game and in programming terms. NPCs won't have to search for each chest in a large room, they just have to search for the much larger shelf, each that can have many individual chests stored.

**Reply by TIKIRobo** [reply]
as an addon to this you could have zones (like in rimworld) where larger cargo is placed, that way a mech/hover/small drone could move up to it and pick it up with out the shelves getting in the way. something like a "flat shelf" would also work here
